File: field-manager-dotnet-backend/Controllers/SeedingController.cs
using FieldManagerDotnetBackend.Models;
using FieldManagerDotnetBackend.Services;
using Microsoft.AspNetCore.Mvc;
using System;

namespace FieldManagerDotnetBackend.Controllers
{
    /// <summary>
    /// Exposes seeding recommendations for a field, representing the
    /// variable rate seeding maps in xarvio FIELD MANAGER. Each field
    /// returns both an average density and a zone-specific rate map.
    /// </summary>
    [ApiController]
    [Route("api/fields/{fieldId:guid}/seeding")]
    public class SeedingController : ControllerBase
    {
        private readonly DemoDataService _dataService;

        public SeedingController(DemoDataService dataService)
        {
            _dataService = dataService;
        }

        [HttpGet]
        public ActionResult<SeedingRecommendation> GetSeeding(Guid fieldId)
        {
            var field = _dataService.GetField(fieldId);
            if (field is null || field.SeedingRecommendation is null)
            {
                return NotFound();
            }
            return Ok(field.SeedingRecommendation);
        }
    }
}

File: fieldmanager-spring-backend/src/main/java/com/xarvio/fieldmanager/controllers/SeedingController.java
package com.xarvio.fieldmanager.controllers;

import java.util.UUID;
import lombok.RequiredArgsConstructor;

import com.xarvio.fieldmanager.models.Field;
import com.xarvio.fieldmanager.models.SeedingRecommendation;
import com.xarvio.fieldmanager.services.DemoDataService;

import org.springframework.web.bind.annotation.RestController;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;

@RestController
@RequiredArgsConstructor
public class SeedingController {

    private final DemoDataService demoDataService;

    @GetMapping("api/fields/{fieldId}/seeding")
    public ResponseEntity<SeedingRecommendation> getSeeding(@PathVariable UUID fieldId) {
        return demoDataService.getField(fieldId)
                .map(Field::seedingRecommendation)
                .map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

}
